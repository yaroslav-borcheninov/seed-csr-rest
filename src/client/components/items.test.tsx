import React from "react"
import { LinkProps } from "react-router-dom"
import { render, RenderResult } from "@testing-library/react"

import Items from "./items"

jest.mock("react-router-dom", () => ({
  Link: ({ to, children }: LinkProps) =>
    `[Link to="${to}" children="${children}"]`,
}))

describe("Items", () => {
  let RealFetch: GlobalFetch["fetch"]
  let MockFetch: jest.Mock
  let renderResult: RenderResult

  beforeEach(() => {
    RealFetch = window.fetch
    MockFetch = jest.fn().mockReturnValue({
      ok: true,
      json: () => [
        {
          id: "123",
          title: "item 123",
        },
        {
          id: "321",
          title: "item 321",
        },
      ],
    })
    window.fetch = MockFetch

    renderResult = render(<Items />)
  })

  afterEach(() => {
    window.fetch = RealFetch
  })

  it("should call fetch", () => {
    expect(MockFetch).toHaveBeenCalledWith("/api/items")
  })

  it("should render data", () => {
    expect(renderResult.container).toMatchSnapshot()
  })
})
