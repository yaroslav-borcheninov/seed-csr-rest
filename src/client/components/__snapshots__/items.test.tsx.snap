// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items should render data 1`] = `
<div>
  <h1
    class="sc-htpNat dpfqCK"
  >
    Items
  </h1>
  <ul>
    <li>
      [Link to="/items/123" children="item 123"]
    </li>
    <li>
      [Link to="/items/321" children="item 321"]
    </li>
  </ul>
</div>
`;
